internal/controller/clusterinstance_controller.go

```diff
@@ -304,20 +309,23 @@ func (r *ClusterInstanceReconciler) handleValidate(
 
 func (r *ClusterInstanceReconciler) renderManifests(
 	ctx context.Context,
+	log *zap.Logger,
 	clusterInstance *v1alpha1.ClusterInstance,
 ) (ci.RenderedObjectCollection, error) {
-	r.Log.Info(fmt.Sprintf("Rendering templates for ClusterInstance %s", clusterInstance.Name))
+	log.Info("Starting to render templates")
 
 	patch := client.MergeFrom(clusterInstance.DeepCopy())
+	r.TmplEngine.SetLogger(log.Named("TemplateEngine"))
```

Comment:
Same here, I don't see why this would be better than sending the logger in the function call.

If you ever try to handle multiple reconcile requests concurrently this will break things (or at least make you log incorrectly).

Reply:
+1